apps/omg_watcher/lib/omg_watcher/db/transaction.ex

```diff
@@ -202,16 +202,16 @@ defmodule OMG.Watcher.DB.Transaction do
     }
   end
 
-  defp filter_constrains(constrains, allowed_constrains) do
-    case Keyword.drop(constrains, allowed_constrains) do
+  defp filter_constraints(constraints, allowed_constraints) do
+    case Keyword.drop(constraints, allowed_constraints) do
       [{out_of_schema, _} | _] ->
         _ =
-          Logger.warn("Constrain on #{inspect(out_of_schema)} does not exist in schema and was dropped from the query")
+          Logger.warn("Constraint on #{inspect(out_of_schema)} does not exist in schema and was dropped from the query")
 
-        constrains |> Keyword.take(allowed_constrains)
+        constraints |> Keyword.take(allowed_constraints)
```

Comment:
`Keyword.take(constraints, allowed_constraints)`

Reply:
the matter of taste I prefer functional approach

Reply:
What makes you think this is functional and the other isn’t?

Reply:
https://elixir-lang.org/getting-started/enumerables-and-streams.html#the-pipe-operator -> `Its purpose is to highlight the data being transformed by a series of functions.`

You're not chaining function calls, you're abusing the operator. Functional or non functional, it has nothing to do with it.